src/Interpreters/Streaming/JoinStreamDescription.h

```diff
@@ -64,11 +68,20 @@ struct JoinStreamDescription
     ///
     /// SELECT * FROM left ASOF JOIN right
     ///    ON left.k = right.k AND left.version < right.version AND
-    ///       lag_behind(left.ts, right.ts, 20)
+    ///       lag_behind(20ms, left.ts, right.ts)
+    /// SETTINGS keep_versions=16;
+    ///
+    /// OR
+    ///
+    /// SELECT * FROM left ASOF JOIN right
+    ///    ON left.k = right.k AND left.version < right.version AND
+    ///       aligned_with(left.ts, right.ts)
```

Comment:
why wen need another function `aligned_with(...)` ? Can we merge the 2 into one ? We will need express the `latency` parameter anyway. The current `aligned_with(left.ts, right.ts)` needs fallback to query settings to refer the join latency.

Reply:
There are 2 options: create a general align function like `align(<what-to-align or align-policy-name>, <latency>, [col, ...])`. Or we can create different align functions for different purposes / scenarios.

If the former, something we will have is like 

```
align('watermark', 20ms, left.ts, right.ts);
align('interleave', 1);  <== interleaved consuming left / right inputs. Not necessary per row, but per chunk.
align(...);
```

BTW, it appears `align(...)` - the function name looks weird in a AND clause

Reply:
or we can introduce another clause.

```
SELECT * FROM left ASOF JOIN right ON left.k = right.k AND left.version < right.version ALIGN ...
```

Reply:
Function seems easier implementation wise.